core/src/main/java/org/mule/runtime/core/internal/policy/DefaultPolicyManager.java

```diff
@@ -207,6 +212,41 @@ public void initialise() throws InitialisationException {
     policyPointcutParametersManager =
         new PolicyPointcutParametersManager(registry.lookupAllByType(SourcePolicyPointcutParametersFactory.class),
                                             registry.lookupAllByType(OperationPolicyPointcutParametersFactory.class));
+
+    // Register flow disposal listener
+    muleContext.getNotificationManager().addListener(new FlowConstructNotificationListener<FlowConstructNotification>() {
+
+      @Override
+      public boolean isBlocking() {
+        return false;
+      }
+
+      @Override
+      public void onNotification(FlowConstructNotification notification) {
+        if (Integer.parseInt(notification.getAction().getIdentifier()) == FLOW_CONSTRUCT_DISPOSED) {
```

Comment:
these would be needed for STOP, not dispose

Reply:
Resolving as discussed. Cache evicting should go in disposal stage.